text/0000-devtools-protocol.md

```diff
@@ -24,44 +24,37 @@ See the current protocol here: https://xstate.js.org/docs/packages/xstate-inspec
 `globalThis.__xstate__` is a global `XStateDevInterface` object:
 
 ```ts
-interface InspectedEventObject {
-  name: string; // Event type
-  data: AnyEventObject; // The actual event object
-  origin?: string; // Session ID
-  destination: string; // Session ID
-  createdAt: number; // Timestamp
-}
-
 interface InspectedActorObject {
   actorRef: AnyActorRef;
   sessionId: string;
   parent?: string; // Session ID
-  snapshot: any; 
+  updates: ActorUpdate[];
   machine?: StateMachineDefinition; // This is originally StateNodeDefinition (renaming)
-  events: InspectedEventObject[];
   createdAt: number; // Timestamp
-  updatedAt: number; // Timestamp
-  status: 0 | 1 | 2; // 0 = not started, 1 = started, 2 = stopped
+  updatedAt: number; // Timestamp, derived from latest update createdAt
+  status: 0 | 1 | 2; // 0 = not started, 1 = started, 2 = stopped, derived from latest update status
 }
 
 interface ActorUpdate {
```

Comment:
Looking at this as the parallel for the `WorkflowEvent` model in our current DB implementation. I have the following notes/discrepancies:
- `ActorUpdate` has the notion of a `sessionId`, while the `WorkflowEvent` does not. We consider each Workflow its own session implicitly. 
- `origin` can be changed to source - they may be interchangeable terms
- `actorRef` and `workflow` may be interchangeable terms for both models. 
- `ActorUpdate` does not have its own `id`

Reply:
> * `ActorUpdate` has the notion of a `sessionId`, while the `WorkflowEvent` does not. We consider each Workflow its own session implicitly.

We need a way to tie the actor "event" to the actor that received it, and the way that we uniquely identify each actor is by the `sessionId` (should we call it `actorId`?)

Reply:
I think `actorId` would be more succinct, yes. The way we do it with the WorkflowEvent is by referencing the `workflowId`, so it would be closer that way I believe.

Reply:
Going to keep it at `sessionId` since "actor ID" is ambiguous with a somewhat-unique identifier for actors used in XState (can explain offline)

Reply:
Understood. We don't need to overload vocabulary if we already use it.